plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1766,15 +1766,15 @@ public function delete( &$order, $args = array() ) {
 
 			$order->set_id( 0 );
 
-			// If this datastore method is called while the posts table is authoritative, refrain from deleting post data.
-			if ( $order->get_data_store()->get_current_class_name() !== self::class ) {
-				return;
+			// If this datastore method is called while the posts table is authoritative, or if data synchronization is disabled,
+			// refrain from deleting post data.
```

Comment:
I found this comment a bit confusing, I think because it's saying kind of the opposite of what the `if` condition is actually doing.

```suggestion
			// Only delete post data if synchronization is enabled, or if the posts table is authoritative.
```

Reply:
The comment is maybe confusing, but it's correct; it's using the fact that "A and B == not((not A) or (not B))". Your comment is better but the "or" should be an "and", I'll change it accordingly.